cvat/apps/dataset_manager/views.py

```diff
@@ -127,14 +156,46 @@ def export_project_annotations(project_id, dst_format=None, server_url=None):
     return export(dst_format, project_id=project_id, server_url=server_url, save_images=False)
 
 
-def clear_export_cache(file_path, file_ctime, logger):
+class FileIsBeingUsedError(Exception):
+    pass
+
+def clear_export_cache(file_path: str, file_ctime: float, logger: logging.Logger) -> None:
+    # file_ctime is for backward compatibility with older RQ jobs, not needed now
+
     try:
-        if osp.exists(file_path) and osp.getctime(file_path) == file_ctime:
+        with get_export_cache_lock(
+            file_path,
+            block=True,
+            acquire_timeout=EXPORT_CACHE_LOCK_TIMEOUT,
+            ttl=rq.get_current_job().timeout,
+        ):
+            if not osp.exists(file_path):
+                raise FileNotFoundError("Export cache file '{}' doesn't exist".format(file_path))
+
+            parsed_filename = parse_export_filename(file_path)
+            cache_ttl = get_export_cache_ttl(parsed_filename.instance_type)
```

Comment:
Hmm... if this is the only thing you need `parsed_filename` for, maybe `parse_export_filename` isn't really needed? Seems like you could just do something like `Path(file_path).parts[-4]`. Or you could pass the `cache_ttl` as a parameter when scheduling the job, so that no parsing of any kind is needed.

Reply:
It has changed several times during editing this PR, so I assume, the function can be nice to have.